sql/moz-fx-data-shared-prod/firefox_desktop_derived/newtab_visits_daily_v2/query.sql

```diff
@@ -0,0 +1,284 @@
+    metrics.boolean.newtab_search_enabled AS newtab_search_enabled,
+    mozfun.newtab.is_default_ui_v1(
+      category,
+      name,
+      extra,
+      metrics.string.newtab_homepage_category,
+      metrics.string.newtab_newtab_category
+    ) AS is_default_ui,
+    category AS event_category,
+    name AS event_name,
+    extra AS event_details,
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.newtab_v1`,
+    UNNEST(events)
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    -- selecting the events we need to compute on
+    AND (
+      (category = 'newtab.search' AND name IN ('issued'))
+      OR (category = 'newtab.search.ad' AND name IN ('click'))
+      OR (
+        category = 'pocket'
+        AND name IN (
+          'click',
+          'impression',
+          'dismiss',
+          'thumb_voting_interaction',
+          'thumb_voting_interaction'
+        )
+      )
+      OR (category = 'topsites' AND name IN ('click', 'impression', 'dismiss'))
+      OR (
+        category = 'newtab'
+        AND name IN (
+          'opened',
+          -- weather
+          'weather_change_display',
+          'weather_open_provider_url',
+          'weather_location_selected',
+          -- wallpaper
+          'wallpaper_click',
+          'wallpaper_category_click',
+          'wallpaper_highlight_cta_click',
+          'wallpaper_highlight_dismissed',
+          -- topic_selection
+          'topic_selection_open',
+          'topic_selection_dismiss',
+          'topic_selection_topics_saved',
+          -- section
+          'sections_block_section',
+          'sections_follow_section',
+          'sections_unblock_section',
+          'sections_unfollow_section',
+          'inline_selection_click'
+        )
+      )
+    )
+)
+SELECT
+  submission_date,
+  client_id,
+  mozfun.map.get_key(event_details, 'newtab_visit_id') AS newtab_visit_id,
+  -- ANY_VALUE: because we do not expect those fields to differ within a visit
+  ANY_VALUE(app_version) AS app_version,
+  ANY_VALUE(os) AS os,
+  ANY_VALUE(channel) AS channel,
+  ANY_VALUE(locale) AS locale,
+  ANY_VALUE(country) AS country,
+  ANY_VALUE(homepage_category) AS homepage_category,
+  ANY_VALUE(newtab_category) AS newtab_category,
+  ANY_VALUE(organic_content_enabled) AS organic_content_enabled,
+  ANY_VALUE(sponsored_content_enabled) AS sponsored_content_enabled,
+  ANY_VALUE(sponsored_topsites_enabled) AS sponsored_topsites_enabled,
+  ANY_VALUE(organic_topsites_enabled) AS organic_topsites_enabled,
+  ANY_VALUE(newtab_search_enabled) AS newtab_search_enabled,
+  LOGICAL_OR(is_default_ui) AS is_default_ui,
+  LOGICAL_OR(event_category = 'newtab' AND event_name = 'opened') AS is_newtab_opened,
+  LOGICAL_OR(event_category = 'newtab.search' AND event_name = 'issued')
+  AND LOGICAL_OR(is_default_ui) AS is_search_issued,
+  LOGICAL_OR(event_category = 'newtab.search.ad' AND event_name = 'click')
+  AND LOGICAL_OR(is_default_ui) AS is_search_ad_click,
```

Comment:
Why are we doing two separate `LOGICAL_OR` and combining with AND?  If `is_default_ui` is separated out like this we risk a scenario where, when we have multiple events being combined, `is_default_ui` is true for a different event than the other statement

Reply:
We are including the `AND LOGICAL_OR(is_default_ui)` so as to make the definition of `is_search_ad_click` (for example) TRUE only when the newtab was opened in default ui

Reply:
This would also be the case with `LOGICAL_OR(event_category = 'newtab.search.ad' AND event_name = 'click' AND is_default_ui)`, but in that case you don't have the issue I outlined in the comment

Reply:
newtab.search.ad click and is_default_ui (newtab open) are seperate events and `UNNESTED` as multiple rows not aligned to one another. The  `AND LOGICAL_OR(is_default_ui)` will aggregate and align the multiple events.

What we are doing here will give us the result we are looking for (as requested by Nadia and Ilana):
the specific activity _with_ the newtab opened in default ui

Reply:
I'm not really following how the `UNNEST` affects things here.  It's basically just a left join, right?

This is the case I'm worried about, if you have the following with the same visit_id:

is_default_ui: True, event_name: impression
is_default_ui: False, event_name: click

The current implementation would produce TRUE, but I think it'd make more sense for it to be FALSE

Reply:
Okay I think I get the salience of the UNNEST now, the idea is that the data I wrote above wouldn't happen because it's a many-to-many join, and that it would actually be:

is_default_ui: True, event_name: impression
is_default_ui: True, event_name: click
is_default_ui: False, event_name: impression
is_default_ui: False, event_name: click

In this case I think `LOGICAL_OR(event_category = 'newtab.search.ad' AND event_name = 'click' AND is_default_ui)`, in which case I think it's still preferable because it requires fewer aggregate operations and is more readable

Reply:
@jaredsnyder I am trying to explain this several times now:

`event_category = 'newtab.search.ad' AND event_name = 'click' AND is_default_ui` do not align after the UNNEST. They happen at different events and render as multiple rows for the same visit.

Try this out for yourself. It should help in your understanding. If you still need help understanding, put some time on my calendar.

Reply:
Working on it now but we cannot merge until I understand what's going on.  This comment is not resolved till then, please stop resolving it.

Reply:
Okay I finally understand what you meant with the non-alignment. The system with two `LOGICAL_OR` is _correct_, I get it now.

Digging into it there are very rare cases that look like this:

is_default_ui: TRUE, event_name: open
is_default_ui: False, event_name: impression
is_default_ui: FALSE, event_name: open
is_default_ui: False, event_name: click

The current logic would give us a TRUE value which I would argue is incorrect.  The easiest fix might be to say any visit with any FALSE values is invalid, so it'd be something like:
`LOGICAL_OR(event_category = 'newtab.search' AND event_name = 'issued')
  AND NOT LOGICAL_OR(NOT is_default_ui) AS is_search_issued`

Reply:
This only works if we can have it be NULL rather than False for events that aren't a tab open though

Reply:
Also please leave a comment here about what's going on with the different events and how that affects the aggregation, I clearly got tripped up on it 😅

Reply:
> Okay I finally understand what you meant with the non-alignment. The system with two LOGICAL_OR is correct, I get it now.

I am glad you were able to do your own analysis to believe how the implementation works.


> Digging into it there are very rare cases that look like this:
is_default_ui: TRUE, event_name: open
is_default_ui: False, event_name: impression
is_default_ui: FALSE, event_name: open
is_default_ui: False, event_name: click

the data models we are building are meant to serve reporting requirements. Not for the purposes of exposing rare data anomalies.

Reply:
I know it is a rare case but a lot of data validation is making sure we are both aware and comfortable with how these cases are handled.

Reply:
> This only works if we can have it be NULL rather than False for events that aren't a tab open though

I do not think we are going to make changes to the udf. However, as mentioned [here](https://github.com/mozilla/bigquery-etl/pull/7362#discussion_r2093745596), would like to understand the use cases associated with the NULLs and the value of those use cases.